docs/madr/decisions/074-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,162 @@
+# Standardized Naming for internal xDS Resources
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/13266
+
+Supersedes: https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/036-internal-listeners.md
+
+## Context and Problem Statement
+
+We use resource names to identify listeners, clusters, routes, virtual hosts and secrets in the xDS APIs.
+These names appear in metrics, logs, and debug endpoints (config_dump), and they are often used for monitoring, filtering, and diagnostics.
+
+It's important to know the difference between system resources (the ones that are defined by Kuma itself, and they usually do not concern the end users) and user resources (the ones that are defined indirectly by the user and the users are interested in).
+For example, clusters like `kuma:envoy:admin` or `_kuma:dynamicconfig` are system resources, while `kri_msvc_mesh-1_us-east-2_kuma-demo_backend_httpport` cluster is user resource.
+
+Currently, there is inconsistency in how these resource names are formed, some of them have a `_kuma` prefix, some of them just contain `kuma` and some are free form.
+
+Some internal resources do not map clearly to Kuma resources in the data store, making it difficult to relate metrics to Kuma abstractions.
+
+There are also resources that seem to be in a bit of a gray area, like `meshtrace_datadog` which is a cluster created for `MeshTrace` policy.
+It could be traced down to MeshTrace resource `kri_mtr_mesh-1__kuma-system_my-meshtrace_` and if a cluster operator has problems with tracing collection it might be something to look at.
+
+To make it easier to distinguish between the two types we introduce the following definition:
+
+- **User resource** - any resource that comes from `Mesh*Service` and `Mesh*Route` definition
```

Comment:
I think in the future we might also have Secrets: since we should probably name it based on MeshIdentity resource. In case of Spire resource name is SPIFFEID

Reply:
I think it's the same thing as a cluster for `MeshTrace` - what do others think?

Reply:
1. SPIRE = no choice (spiffe ID / trust Domain)
2. Not SPIRE
  1. we can't do KRI because it's a combo of multiple all the MeshTrusts
  2. For MeshIdentity we could use KRI (future)
  3. For current `system_<category>_<scope>_<identifier>
  4. For user defined secrets we can use KRI